CorsixTH/Lua/rooms/ward.lua

```diff
@@ -123,36 +124,63 @@ function WardRoom:doStaffUseCycle(humanoid)
   humanoid:queueAction(MeanderAction():setLoopCallback(meanders_loop))
 end
 
+function WardRoom:calculateHealingFactor()
+  local patient_count = 1
+  local nurse_factor = 0
+  for humanoid in pairs(self.humanoids) do
+    if not humanoid:isLeaving() then
+      if class.is(humanoid, Patient) then
+        patient_count = patient_count + 1
+      elseif humanoid.humanoid_class == "Nurse" and not humanoid.fired then
+        nurse_factor = nurse_factor + 0.5 + humanoid:getServiceQuality()
+      end
+    end
+  end
+  -- Difficulty of healing - 10% faster on easier, 10% slower on hard mode
+  local difficulty_factor = 0.8 + self.world.map:getDifficulty() * 0.1
+  if nurse_factor > 0 then
+    self.healing_amount = nurse_factor / math.log(patient_count) / difficulty_factor
+  else
+    self.healing_amount = 0
+  end
+end
+
 
--- TODO the nurse should not leave the ward if there are beds in use, therefore prevent her from being picked up
--- and have a system that stops patients entering the ward if she is in need of taking a break or being called elsewhere.
 function WardRoom:commandEnteringPatient(patient)
-  local staff = next(self.staff_member_set) or self.staff_member
   local bed, pat_x, pat_y = self.world:findFreeObjectNearToUse(patient, "bed")
   self:setStaffMembersAttribute("dealing_with_patient", nil)
   if not bed then
     patient:setNextAction(self:createLeaveAction())
     patient:queueAction(self:createEnterAction(patient))
     print("Warning: A patient was called into the ward even though there are no free beds.")
```

Comment:
Outside of scope but if we can could we change this to a gamelog rather than a print? Though depends if it'll get in the way of #1833

Reply:
Done, I'll fix it in that branch.